tool/plugin/test/plugin_test.dart

```diff
@@ -18,24 +16,25 @@ void main() {
       expect(new GenCommand(new BuildCommandRunner()).name, "gen");
     });
   });
+
   group("spec", () {
     test('build', () {
       var runner = new BuildCommandRunner();
-      var cmd = new AntBuildCommand(runner);
+      var cmd = new TestAntBuildCommand(runner);
       runner.addCommand(cmd);
-      var args = runner.parse(["-r=19", "build"]);
-      var mockCmd = new MockCommand();
-      when(mockCmd.argResults).thenReturn(args);
-      // Apparently spy() does not work the way I want it to.
-      if (0 == 0) return;
-      var specs = createBuildSpecs(spy(mockCmd, cmd));
-      expect(specs.length, 2);
-      expect(specs[0].ideaProduct, 'android-studio-ide');
-      expect(specs[1].ideaProduct, 'ideaIC');
+      runner.run(["-r=19", "build"])
+          .whenComplete(() {
+        var specs = cmd.specs;
+        expect(specs.length, 2);
+        expect(specs[0].ideaProduct, 'android-studio-ide');
+        expect(specs[1].ideaProduct, 'ideaIC');
```

Comment:
You might also try something like:

    expect(specs, orderedEquals(['android-studio-ide', 'ideaIC']));

and delete the previous 3 lines.

Reply:
Good idea, thanks!